diff.go

```diff
@@ -268,14 +283,15 @@ func (p *diffParser) parseFileHeader() (*DiffFile, error) {
 	}
 
 	file := &DiffFile{
-		Name: a,
-		Type: DiffFileChange,
+		Name:    a,
+		oldName: b,
+		Type:    DiffFileChange,
 	}
 
 	// Check file diff type and submodule
 	var err error
 checkType:
-	for !p.isEOF {
+	for {
```

Comment:
Can you elaborate why is this a necessary change?

Reply:
Hmm, good catch, I think this is not necessary.